src/cascadia/QueryExtension/OpenAILLMProvider.cpp

```diff
@@ -0,0 +1,126 @@
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::System;
+namespace WWH = ::winrt::Windows::Web::Http;
+namespace WSS = ::winrt::Windows::Storage::Streams;
+namespace WDJ = ::winrt::Windows::Data::Json;
+
+static constexpr std::wstring_view applicationJson{ L"application/json" };
+static constexpr std::wstring_view acceptedModel{ L"gpt-3.5-turbo" };
+static constexpr std::wstring_view openAIEndpoint{ L"https://api.openai.com/v1/chat/completions" };
+
+namespace winrt::Microsoft::Terminal::Query::Extension::implementation
+{
+    OpenAILLMProvider::OpenAILLMProvider(const winrt::hstring& key)
+    {
+        _AIKey = key;
+        _httpClient = winrt::Windows::Web::Http::HttpClient{};
+        _httpClient.DefaultRequestHeaders().Accept().TryParseAdd(applicationJson);
+        _httpClient.DefaultRequestHeaders().Authorization(WWH::Headers::HttpCredentialsHeaderValue{ L"Bearer", _AIKey });
+    }
+
+    void OpenAILLMProvider::ClearMessageHistory()
+    {
+        _jsonMessages.Clear();
+    }
+
+    void OpenAILLMProvider::SetSystemPrompt(const winrt::hstring& systemPrompt)
+    {
+        WDJ::JsonObject systemMessageObject;
+        winrt::hstring systemMessageContent{ systemPrompt };
+        systemMessageObject.Insert(L"role", WDJ::JsonValue::CreateStringValue(L"system"));
+        systemMessageObject.Insert(L"content", WDJ::JsonValue::CreateStringValue(systemMessageContent));
+        _jsonMessages.Append(systemMessageObject);
+    }
+
+    void OpenAILLMProvider::SetContext(const Extension::IContext context)
+    {
+        _context = context;
+    }
+
+    winrt::Windows::Foundation::IAsyncOperation<Extension::IResponse> OpenAILLMProvider::GetResponseAsync(const winrt::hstring userPrompt)
+    {
+        // Use a flag for whether the response the user receives is an error message
+        // we pass this flag back to the caller so they can handle it appropriately (specifically, ExtensionPalette will send the correct telemetry event)
+        // there is only one case downstream from here that sets this flag to false, so start with it being true
+        bool isError{ true };
+        hstring message{};
+
+        // Make sure we are on the background thread for the http request
+        co_await winrt::resume_background();
+
+        WWH::HttpRequestMessage request{ WWH::HttpMethod::Post(), Uri{ openAIEndpoint } };
+        request.Headers().Accept().TryParseAdd(applicationJson);
+
+        WDJ::JsonObject jsonContent;
+        WDJ::JsonObject messageObject;
+
+        winrt::hstring engineeredPrompt{ userPrompt };
+        if (_context && !_context.ActiveCommandline().empty())
+        {
+            engineeredPrompt = userPrompt + L". The shell I am running is " + _context.ActiveCommandline();
```

Comment:
I need to debug more through this, but have you seen any issues with the active command line? This should be pwsh.exe, I think. 

![image](https://github.com/user-attachments/assets/60720971-612c-4f09-871f-96a338f6a76b)

Reply:
Yes this is a bug right now with how we try to obtain the commandline exe, I'll fix that in a PR separately